### PR TITLE
Add new endpoints (teams and check user sub)

### PR DESCRIPTION
**Pull request by ThibaultVlacich**
This PR adds the new endpoints announced today.

Announcement: https://discuss.dev.twitch.tv/t/the-latest-twitch-api-endpoints-for-twitch-teams-and-subscriptions/30838

### CODE REVIEW

**Comment by brandinarsenault** [comment]
@ThibaultVlacich thank you for your contribution. Please complete the following requests:

- Consider adding a helper function for `getTeams`: `getTeamByName` `getTeamById` which would both only accept 1 parameter and then supply it to `getTeams` (optional)
- Add the specs for all of the endpoints you added as these are required for tests and must pass for us to merge the new endpoints you've submitted.

Otherwise I see no issues with your PR so far.

**Comment by ThibaultVlacich** [comment]
@brandinarsenault Done. There was no Spec for the already existing `SubscriptionsApi`, so I added it but only added the one endpoint I'm adding in this PR, to avoid including unrelated things in this PR.

**Review by brandinarsenault — CHANGES_REQUESTED** [review]

**Comment by ThibaultVlacich** [comment]
@brandinarsenault Weird crash on the CI, it's showing a git diff?

File: src/NewTwitchApi/Resources/TeamsApi.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            $queryParamsMap[] = ['key' => 'name', 'value' => $name];
+        }
+
+        if ($id) {
+            $queryParamsMap[] = ['key' => 'id', 'value' => $id];
+        }
+
+        return $this->getApi('teams', $bearer, $queryParamsMap);
+    }
+
```
Comment (brandinarsenault) [inline src/NewTwitchApi/Resources/TeamsApi.php:43]: Theres some kind of weird tab or space here causing the Coding Standards check to fail. Just run `php-cs-fixer fix` on this file and it should clean it up for you and let the test pass.

**Review by brandinarsenault — CHANGES_REQUESTED** [review]

File: src/NewTwitchApi/Resources/TeamsApi.php (hunk 1 above)
Comment (ThibaultVlacich) [reply src/NewTwitchApi/Resources/TeamsApi.php]: Oh I see

**Review by brandinarsenault — APPROVED** [review]

**Comment by brandinarsenault** [comment]
@ThibaultVlacich thank you! :)

**Pull request merged**